samples/testapp/src/commonMain/kotlin/com/android/identity/testapp/presentation/PresentationModel.kt

```diff
@@ -0,0 +1,146 @@
+package com.android.identity.testapp.presentation
+
+import com.android.identity.util.Logger
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+// TODO: move this entire package to identity-appsupport when it's fully baked.
+
+/**
+ * A model used for credential presentations.
+ *
+ * This is a singleton, use [getInstance] to obtain a reference.
+ */
+class PresentationModel {
+
+    companion object {
+        private const val TAG = "PresentationModel"
+
+        private val _instance = PresentationModel()
+
+        /**
+         * Gets the singleton instance.
+         */
+        fun getInstance(): PresentationModel = _instance
+    }
+
+    /**
+     * Possible states that the model can be in.
+     */
+    enum class State {
+        /**
+         * Presentation is not active.
+         */
+        IDLE,
+
+        /**
+         * A presentation has been started but the mechanism used to communicate with the credential reader is not yet
+         * available.
+         */
+        WAITING,
+
+        /**
+         * A presentation is currently underway and the mechanism to communicate with the credential reader is
+         * available.
+         */
+        RUNNING,
+
+        /**
+         * A presentation has been completed.
+         */
+        COMPLETED,
+    }
+
+    private val _state = MutableStateFlow<State>(State.IDLE)
+
+    /**
+     * The current state of the presentation model.
+     */
+    val state: StateFlow<State> = _state.asStateFlow()
```

Comment:
Rename to stateFlow, for clarity?

Reply:
Will consider that for next PR (it has a lot of `StateFlow<>` properties so might be tedius).